pkg/controller/registrysyncer/registrysyncer_test.go

```diff
@@ -0,0 +1,454 @@
+						},
+					},
+				}
+				if diff := cmp.Diff(expectedImageStream, expectedImageStream); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name: "app.ci is newer",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: bcc(fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTag.DeepCopy()}...)),
+			appCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTagNewer.DeepCopy(), applyconfigIS.DeepCopy()}...),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				actualImageStreamImport := &imagev1.ImageStreamImport{}
+				if err := apiCIClient.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStreamImport); err != nil {
+					return err
+				}
+				expectedImageStreamImport := &imagev1.ImageStreamImport{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "ImageStreamImport",
+						APIVersion: "image.openshift.io/v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace:       "ci",
+						Name:            "applyconfig",
+						ResourceVersion: "1",
+					},
+					Spec: imagev1.ImageStreamImportSpec{
+						Import: true,
+						Images: []imagev1.ImageImportSpec{{
+							From: corev1.ObjectReference{
+								Kind: "DockerImage",
+								Name: "registry.ci.openshift.org/ci/applyconfig@sha256:new",
+							},
+							To: &corev1.LocalObjectReference{Name: "latest"},
+							ReferencePolicy: imagev1.TagReferencePolicy{
+								Type: imagev1.LocalTagReferencePolicy,
+							},
+						}},
+					},
+					Status: imagev1.ImageStreamImportStatus{
+						Images: []imagev1.ImageImportStatus{
+							{
+								Image: &imagev1.Image{},
+							},
+						},
+					},
+				}
+				if diff := cmp.Diff(expectedImageStreamImport, actualImageStreamImport); diff != "" {
+					return fmt.Errorf("actual does not match expected, diff: %s", diff)
+				}
+				return nil
+			},
+		},
+		{
+			name: "app.ci is newer but refers to the same image",
+			request: types.NamespacedName{
+				Name:      "applyconfig:latest",
+				Namespace: "ci",
+			},
+			apiCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTag.DeepCopy()}...),
+			appCIClient: fakeclient.NewFakeClientWithScheme(scheme.Scheme, []runtime.Object{applyconfigISTagNewerSameName.DeepCopy(), applyconfigIS.DeepCopy()}...),
+
+			verify: func(apiCIClient ctrlruntimeclient.Client, appCIClient ctrlruntimeclient.Client) error {
+				for clusterName, client := range map[string]ctrlruntimeclient.Client{apiCI: apiCIClient, appCI: appCIClient} {
+					actualImageStreamImport := &imagev1.ImageStreamImport{}
+					err := client.Get(ctx, types.NamespacedName{Name: "applyconfig", Namespace: "ci"}, actualImageStreamImport)
+					if !errors.IsNotFound(err) {
```

Comment:
Nit: You could `list` here and verify len(result)  == 0, that would make the test correctly fail even if the import had a different name

Reply:
Left a comment to address why we cannot use `List` here.